components/salsa-macro-rules/src/setup_accumulator_impl.rs

```diff
@@ -34,9 +34,13 @@ macro_rules! setup_accumulator_impl {
                 static $CACHE: $zalsa::IngredientCache<$zalsa_struct::IngredientImpl<$Struct>> =
                     $zalsa::IngredientCache::new();
 
-                $CACHE.get_or_create(zalsa, || {
-                    zalsa.lookup_jar_by_type::<$zalsa_struct::JarImpl<$Struct>>()
-                })
+                // SAFETY: `lookup_jar_by_type` returns a valid ingredient index, and the only
+                // ingredient created by our jar is the struct ingredient.
+                unsafe {
```

Comment:
I think it's technically possible that a salsa user implements their own `Ingredient`. Could a malicious `Ingredient` implementation invalidate the SAFETY assumption?

Reply:
No, this just accesses the ingredient created by this jar.